llvm/lib/Target/RISCV/RISCV.td

```diff
@@ -369,6 +369,15 @@ def FeatureRV32E
 def IsRV32E : Predicate<"Subtarget->isRV32E()">,
                         AssemblerPredicate<(all_of FeatureRV32E)>;
 
+// TODO: Once all supported CheriBSD branches are ready for the new semantics
+// this feature should be enabled automatically.
+def FeatureNoCheriV8Compat
+    : SubtargetFeature<"xcheri-no-v8-compat", "HasCheriISAv8Semantics", "false",
+                       "CHERI ISAv8 semantics (trapping, DDC/PCC relocation)">;
+def HasCheriISAv8
```

Comment:
I think it would be nice to export some sort of `#define` for this so that we can use `#ifdef` in the few places in CheriBSD (e.g. cheribsdtest, but also __USER_CAP) that care about DDC/PCC offsetting and tag clearing (the onfault wrapper around CBuildCap).

Reply:
Yes I was planning on adding that. Something like `__riscv_xcheri_tag_clearing` and `__riscv_xcheri_no_relocation`? Or just a single one?

Reply:
I was kind of thinking of something more like `__CHERI_ISA_VERSION` or the like that was the major version of the corresponding ISA spec?  Maybe that's too hard to be viable as a cross-architecture one though vs having individual ones like you suggest.  For RISC-V at least I feel like we could maybe have a single one for all of the version 9 changes, so maybe a `__riscv_xcheri_version` or `__riscv_xcheri_isa_version`.  One could also suggest that maybe we should somehow morph this into the version of the `xcheri` RISC-V extension in the ISA string, but maybe we can avoid that since we are using bits in xccsr instead.

Reply:
Due to the way LLVM deals with versioned extensions (effectively there can only be one supported version right now), I did not use the versioning approach.